records_mover/records/job/config.py

```diff
@@ -72,24 +77,43 @@ def fill_in_existing_table_handling(self, kwargs: Dict[str, Any]) -> None:
 
-    def fill_in_records_format(self, kwargs: Dict[str, Any]) -> None:
-        if kwargs['variant'] is not None:
-            if 'records_format' in kwargs:
-                # We've already started filling this out with hints -
-                # but now we know which variant they want
-                kwargs['records_format'] = kwargs['records_format'].alter_variant(kwargs['variant'])
-            else:
-                hints: PartialRecordsHints = {}
-                if 'initial_hints' in kwargs:
-                    # Given the user gave us a variant, we won't be
-                    # using "initial hints" to sniff with - the hint
-                    # parameters they gave us should be specified as
-                    # part of the records format instead.
-                    hints = kwargs['initial_hints']
-                    del kwargs['initial_hints']
-                kwargs['records_format'] = DelimitedRecordsFormat(variant=kwargs['variant'],
-                                                                  hints=hints)
-            del kwargs['variant']
+    def fill_in_format(self, kwargs: Dict[str, Any]) -> None:
+        records_format: BaseRecordsFormat
+        if kwargs['format'] == 'delimited':
+            records_format = DelimitedRecordsFormat()
+        elif kwargs['format'] == 'parquet':
+            records_format = ParquetRecordsFormat()
+        else:
+            raise NotImplementedError(f"No such records format type: {kwargs['format']}")
+
+        if 'records_format' in kwargs:
+            existing_records_format = kwargs['records_format']
+            if type(records_format) != type(existing_records_format):
+                raise NotImplementedError('Hints are not compatible '
+                                          'with records format '
+                                          f'{existing_records_format.format_type}')
+        else:
+            kwargs['records_format'] = records_format
+
+        del kwargs['format']
+
+    def fill_in_variant(self, kwargs: Dict[str, Any]) -> None:
+        if 'records_format' in kwargs:
+            # We've already started filling this out with hints -
+            # but now we know which variant they want
+            kwargs['records_format'] = kwargs['records_format'].alter_variant(kwargs['variant'])
+        else:
+            hints: PartialRecordsHints = {}
+            if 'initial_hints' in kwargs:
+                # Given the user gave us a variant, we won't be
+                # using "initial hints" to sniff with - the hint
+                # parameters they gave us should be specified as
+                # part of the records format instead.
+                hints = kwargs['initial_hints']
+                del kwargs['initial_hints']
+            kwargs['records_format'] = DelimitedRecordsFormat(variant=kwargs['variant'],
+                                                              hints=hints)
+        del kwargs['variant']
```

Comment:
We now have both a `format` and `variant` command line parameter, instead of just `variant`.  These functions handle ingesting two two, respectively.

The 'initial_hints' handling is not new or changed with this PR.  It deals with the situation where we want RecordsMover to generally sniff the records format, but specify certain hints to begin with as constraints.